config/routes.rb

```diff
@@ -1,9 +1,17 @@
 Rails.application.routes.draw do
-  get 'users/new'
-  get 'carts/show'
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
   root 'products#index'
   get '/dashboard', to: 'dashboard#index'
+
+  get '/signup', to: 'users#new'
+  get '/login',   to: 'sessions#new'
+  post   '/login',   to: 'sessions#create'
+  delete '/logout',  to: 'sessions#destroy'
+
+  resources :products do
+    post :add_to_cart, on: :member
+    delete :remove_from_cart, on: :member
+  end
   
   resources :products
```

Comment:
you already have `resources :products` above. if you need the `new, index, create, etc..` methods, you could do:
```ruby
resources :products, only: %i[new create index] do
  post :add_to_cart, on: :member
  delete :remove_from_cart, on: :member
end
```